src/plugins/simulator/media/rab_medium.cpp

```diff
@@ -42,9 +42,9 @@ namespace argos {
          if(strPosIndexMethod == "grid") {
             size_t punGridSize[3];
             if(!NodeAttributeExists(t_tree, "grid_size")) {
-               punGridSize[0] = cArenaSize.GetX();
-               punGridSize[1] = cArenaSize.GetY();
-               punGridSize[2] = cArenaSize.GetZ();
+               punGridSize[0] = static_cast<UInt32>(cArenaSize.GetX());
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary? Also, this cast is wrong because it loses the sign.

Reply:
I did a 'UInt32' cast because the destination variable is unisgned, so even if I cast to `SInt32`, the sign will be lost when the variable is stored. Grid sizes should always be non-negative I would think, so this is appropriate?

Reply:
@ilpincy See @allsey87 's reasoning for the `tag_medium.cpp` casts, which I think applies here too.